src/openrct2/paint/Paint.cpp

```diff
@@ -84,21 +84,8 @@ static paint_struct* sub_9819_c(
     paint_struct* ps = &session->NextFreePaintStruct->basic;
     ps->image_id = image_id;
 
-    switch (session->CurrentRotation)
-    {
-        case 0:
-            rotate_map_coordinates(&offset.x, &offset.y, 0);
-            break;
-        case 1:
-            rotate_map_coordinates(&offset.x, &offset.y, 3);
-            break;
-        case 2:
-            rotate_map_coordinates(&offset.x, &offset.y, 2);
-            break;
-        case 3:
-            rotate_map_coordinates(&offset.x, &offset.y, 1);
-            break;
-    }
+    uint8_t swappedRotation = (session->CurrentRotation * 3) % 4; // swaps 1 and 3
```

Comment:
3 * 3 = 9 % 4 = 2, Are you sure this is correct?

Reply:
I think this is correct:
(0 * 3 = 0) mod 4 = 0
(1 * 3 = 3) mod 4 = 3
(2 * 3 = 6) mod 4 = 2
(3 * 3 = 9) mod 4 = 1

Reply:
My bad, I have no idea how I got 2 out of it.